src/base/net/downloadhandlerimpl.cpp

```diff
@@ -121,9 +121,13 @@ void DownloadHandlerImpl::processFinishedDownload()
     }
 
     // Success
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 3, 0))
+    m_result.data = m_reply->readAll();
+#else
     m_result.data = (m_reply->rawHeader("Content-Encoding") == "gzip")
```

Comment:
Assuming it is decompressed by default, the reply should fail this check and the second branch should be taken, which is identical to this, no? This just removes an if check, which might be a better code path for 6.3.0, but this shouldn't "fix" anything

Reply:
I modified the patch to print the raw headers and it seems content encoding is still set to "gzip".

```
May 27 18:22:50 bob qbittorrent-nox[43301]: rawHeader("Content-Encoding") == "gzip": 1
May 27 18:22:50 bob qbittorrent-nox[43301]: Key content-type, Value application/xml
May 27 18:22:50 bob qbittorrent-nox[43301]: Key Content-Length, Value 2965
May 27 18:22:50 bob qbittorrent-nox[43301]: Key content-encoding, Value gzip
```

Reply:
What does it shows when you make it fail the test, for example.

`#if (QT_VERSION >= QT_VERSION_CHECK(6, 4, 0))`

Reply:
> I modified the patch to print the raw headers and it seems content encoding is still set to "gzip".

Don't think it is expected that received header is modified when reading the reply data.